tests/bluetooth/host/buf/bt_buf_get_cmd_complete/src/main.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <ztest.h>
+#include "kconfig.h"
+#include "hooks.h"
+#include "buf_help_utils.h"
+
+#include <zephyr/bluetooth/hci.h>
+#include <host/hci_core.h>
+#include <zephyr/bluetooth/buf.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/slist.h>
+#include <zephyr/sys/util.h>
+
+/*
+ *  Return value from bt_buf_get_cmd_complete() should be NULL
+ *
+ *  This is to test the behaviour when memory allocation request fails
+ *
+ *  Constrains:
+ *     - bt_dev.sent_cmd value is NULL
+ *     - Timeout value is a positive non-zero value
+ *     - net_buf_alloc_fixed() returns a NULL value
+ *
+ *  Expected behaviour:
+ *     - net_buf_alloc_fixed() to be called with the correct memory allocation pool
+ *       and the same timeout value passed to bt_buf_get_cmd_complete()
+ *     - bt_buf_get_cmd_complete() returns NULL
+ *
+ */
+void test_returns_null_sent_cmd_is_null(void)
+{
+	struct net_buf *buf;
+	k_timeout_t timeout = Z_TIMEOUT_TICKS(1000);
+
+	bt_dev.sent_cmd = NULL;
+
+#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_evt_pool());
+#else
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_hci_rx_pool());
+#endif
```

Comment:
We should try to make the branching (including ifdefs) as simple as possible in test. Here we can define a variable `const expected_pool` that get's its value set depending on the kconfig.

```suggestion
#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
	const struct net_buf_pool *expected_pool = bt_buf_get_evt_pool();
#else
	const struct net_buf_pool *expected_pool = bt_buf_get_evt_pool();
#endif
ztest_expect_value(net_buf_alloc_fixed, pool, expected_pool);
```